File: IATA.R
#Downloading and cleaning IATA Data

library(dplyr)
library(curl)
library(httr)
library(jsonlite)
library(tidyr)
library(data.table)

url <- "https://aviation-edge.com/v2/public/airlineDatabase?key=e5c06d-d3426f"
#Getting data from Aviation Edge API
differentAirlines <- fromJSON(url)
codeIata <- c("OS", "EY", "KL", "LH", "QR", "SQ", "SA", "TK")
alt_result <- differentAirlines %>% filter(codeIataAirline%in%codeIata)
alt_result[alt_result==""]<-NA
alt_result <- alt_result %>% drop_na() %>% select(nameAirline, codeIataAirline, sizeAirline, ageFleet, codeIso2Country, nameCountry)
names(alt_result) <- c("Airline", "IATA Code", "Number of Airplanes","Fleet Age(AVG)", "Country Code", "Country")
alt_result[1,1] <- "Turkish Airlines"
alt_result[2,1] <- "Lufthansa"
alt_result[6,1] <- "Singapore Airlines"
alt_result[7,1] <- "Austrian Airlines"
alt_result[8,1] <- "South African Airways"
alt_result <- alt_result %>% arrange(Airline) 
setDT(alt_result)[, AirlineID := .GRP, by = Airline]
setDT(alt_result)[, CountryID := .GRP, by = Country]
alt_result <- alt_result %>% arrange(Airline) %>% select(AirlineID, Airline, CountryID, Country,  `IATA Code`, `Number of Airplanes`, `Fleet Age(AVG)`, `Country Code`)
write.csv(alt_result, file="C:\\Users\\vikra\\OneDrive - National College of Ireland\\MSc Data Analytics\\Data Warehousing and Bussiness Intelligence\\Project\\Final CSVs\\Companies.csv", row.names = F)

File: IATA.py
# import json
# import requests
# headers = {
#     'API-KEY': 'e5c06d-d3426f',
# }
# r = requests.get('https://aviation-edge.com/v2/public/airplaneDatabase?', headers=headers)

import subprocess
import json
import pandas as pd
import feather

# proc = subprocess.run(["curl",  "-X", "GET",
#                   "https://aviation-edge.com/v2/public/planeTypeDatabase?key=e5c06d-d3426f&numberRegistration=1000"],
#                    stdout=subprocess.PIPE, encoding='utf-8')
#
# cadastro = proc.stdout
# planeTypeDatabase = pd.DataFrame([json.loads(cadastro)])
# print(planeTypeDatabase.to_string())
# pd.DataFrame.info(planeTypeDatabase)
# path = 'C:\\Users\\vikra\\OneDrive - National College of Ireland\MSc Data Analytics\\Data Warehousing and Bussiness Intelligence\\Project\\IATA_df.feather'
# feather.write_dataframe(df, path)

get_aviationData = subprocess.run(["curl",  "-X", "GET",
                  "https://aviation-edge.com/v2/public/airlineDatabase?key=e5c06d-d3426f&codeIataAirline=SQ&codeIataAirline=AA&codeIataAirline=QR&codeIataAirline=LH&codeIataAirline=SA&codeIataAirline=EY&codeIataAirline=A3&codeIataAirline=OS&codeIataAirline=QS&codeIataAirline=KM&codeIataAirline=VS&codeIataAirline=KL&codeIataAirline=DY&codeIataAirline=TK&codeIataAirline=GA&codeIataAirline=UA&codeIataAirline=FR&limit=5000"],
                   stdout=subprocess.PIPE, encoding='utf-8')

aviationDataToJSON = get_aviationData.stdout
airlinesDatabase = pd.DataFrame([json.loads(aviationDataToJSON)])
print(airlinesDatabase.to_string())
pd.DataFrame.info(airlinesDatabase)

path = 'C:\\Users\\vikra\\OneDrive - National College of Ireland\MSc Data Analytics\\Data Warehousing and Bussiness Intelligence\\Project\\airlinesDatabase.feather'
feather.write_dataframe(airlinesDatabase, path)

